### PR TITLE
chm-417 financial report 2020 view page

### PR DESCRIPTION
**Pull request by liyuanmontreal**
### General description
migrate financial framework report 2020 view page

### Designs
same as absch project / old chm project

### Testing instructions
search -> financial framework report 2020

remain issue
table in section 5 not tested because loadBaselineDocuments function not finished.

### CODE REVIEW

File: app/views/forms/view/chm/financial-report-2020.vue [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+   })
+   const document = computed(()=>props.documentInfo?.body);
+
+   onMounted(() => {
+       loadBaselineDocuments();
+   })
+
+   const options  = {
+        multipliers : 		[{identifier:'units',	      title: {en:'in units'}},   		   {identifier:'thousands', title: {en:'in thousands'}}, 		{identifier:'millions', title: {en:'in millions'}}],
+        methodology : 		[{identifier:'oecd_dac',      title: {en:'OECD DAC Rio markers'}}, {identifier:'other', 	title: {en:'Other'       }}],
```
Comment (tahmeedshah) [inline app/views/forms/view/chm/financial-report-2020.vue:249]: move the text to translation file.

File: app/views/forms/view/chm/financial-report-2020.vue [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+
+        const query = {
+                $filter : `(type eq '${encodeURI('resourceMobilisation')}')`,  
+                $top: 10,
+                collection: 'my'
+        }
+        
+        let queryDraft = {
+                $filter : `(type eq '${encodeURI('resourceMobilisation')}')`,
+                $top: 10,
```
Comment (tahmeedshah) [inline app/views/forms/view/chm/financial-report-2020.vue:478]: why do you need draft records ?

File: app/views/forms/view/chm/financial-report-2020.vue (hunk 1 above)
Comment (liyuanmontreal) [reply app/views/forms/view/chm/financial-report-2020.vue]: fixed

File: app/views/forms/view/chm/financial-report-2020.vue (hunk 2 above)
Comment (liyuanmontreal) [reply app/views/forms/view/chm/financial-report-2020.vue]: remove code relate to draft record

File: app/views/forms/view/chm/financial-report-2020.vue [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+            var estimate = _.findWhere(estimates, {year:year});
+            if(estimate)
+                return true;
+        }
+        return false;
+    };
+  
+    const loadBaselineDocuments = async function(){
+        const query = {
+            $filter : `(type eq '${encodeURI('resourceMobilisation')}')`
```
Comment (blaisefonseca) [inline app/views/forms/view/chm/financial-report-2020.vue:407]: why encodeURI?

File: app/views/forms/view/chm/financial-report-2020.vue [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+        return false;
+    };
+  
+    const loadBaselineDocuments = async function(){
+        const query = {
+            $filter : `(type eq '${encodeURI('resourceMobilisation')}')` 
+        }
+
+        const records =  new Promise(function(resolve, reject) {
+            kmDocumentApi.queryDocuments(query).then(function(results) {
```
Comment (blaisefonseca) [inline app/views/forms/view/chm/financial-report-2020.vue:411]: what is the purpose of this call? this query will get only the loggedin users records

File: app/views/forms/view/chm/financial-report-2020.vue [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+                else {
+                    reject (e);
+                }
+            });
+        });
+    }	
+
+    const getBaselineDocument = function(identifier) {
+        if (identifier) { //lookup single record        
+            return new Promise(function(resolve, reject) {
```
Comment (blaisefonseca) [inline app/views/forms/view/chm/financial-report-2020.vue:440]: why are async calls wrapped in promise

File: app/views/forms/view/record-loader.vue [hunk 6]
```diff
@@ -9,6 +9,8 @@
                     :document-info="documentInfo" :locale="locale"></national-report>
             <national-target v-if="documentInfo?.type == 'nationalTarget'"  
                     :document-info="documentInfo" :locale="locale"></national-target>
+            <financial-report-2020 v-if="documentInfo?.type == 'resourceMobilisation2020'"  
+                    :document-info="documentInfo" :locale="locale"></financial-report-2020 >
```
Comment (blaisefonseca) [inline app/views/forms/view/record-loader.vue:13]: do we know why we did not add `view-` to the chm view form components like we have for abs and bch

File: app/views/forms/view/chm/financial-report-2020.vue (hunk 3 above)
Comment (liyuanmontreal) [reply app/views/forms/view/chm/financial-report-2020.vue:407]: in old code , use encodeURI
image

**Comment by liyuanmontreal** [comment]
image

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**